app/models/hostgroup.rb

```diff
@@ -212,6 +213,10 @@ def update_ancestry_puppetclasses
     unscoped_find(ancestry.to_s.split('/').last.to_i).update_puppetclasses_total_hosts if ancestry.present?
   end
 
+  def children_hosts_count
+    children.map(&:hosts_count).inject{| x | + x} if children.any?
```

Comment:
I think this should also include the hostgroup's *own* hosts_count, so it becomes a count of all hosts within this hierarchy.  It'll then match the results when you click the search link.  You can change children to subtree, which will include itself, plus it'll include children recursively and not just the first level.

Nitpick, `.inject(&:+)`